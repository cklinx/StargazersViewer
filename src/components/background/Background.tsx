import React, {memo, ReactNode} from 'react';
import {ImageBackground, StyleSheet, KeyboardAvoidingView} from 'react-native';
import useColorScheme from '@stargazers/hooks/useColorScheme';

interface BackgroundProps {
  children: ReactNode;
  style?: {};
}

const BackgroundComponent = ({children, style}: BackgroundProps) => {
  const colorScheme = useColorScheme();
  const src =
    colorScheme === 'dark'
      ? require('../../assets/splashscreen/splashscreen-dark.png')
      : require('../../assets/splashscreen/splashscreen.png');
  return (
    <ImageBackground
      source={src}
      resizeMode="cover"
      style={{
        ...styles.background,
        ...style,
      }}>
      <KeyboardAvoidingView style={styles.container} behavior="padding">
        {children}
      </KeyboardAvoidingView>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  background: {
    flex: 1,
    height: '100%',
    width: '100%',
    position: 'relative',
    backgroundColor: 'white',
  },
  container: {
    flex: 1,
    padding: 0,
    width: '100%',
    alignSelf: 'center',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export const Background = memo(BackgroundComponent);
