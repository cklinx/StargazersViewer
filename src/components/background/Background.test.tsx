import React from 'react';
import {View} from 'react-native';
import renderer from 'react-test-renderer';
import {Background} from './Background';

describe('Background', () => {
  it('should render correctly', () => {
    const {toJSON} = renderer.create(
      <Background>
        <View></View>
      </Background>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
