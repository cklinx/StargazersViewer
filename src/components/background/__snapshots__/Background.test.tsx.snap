// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background should render correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": "100%",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="cover"
    source="test-file-stub"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "padding": 0,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View />
  </View>
</View>
`;
